packages/calypso-e2e/src/lib/blocks/block-components/template-part-block.ts

```diff
@@ -0,0 +1,44 @@
+import { Locator, Page } from 'playwright';
+
+const selectors = {
+	chooseButton: 'button:has-text("Choose")',
+	startBlankButton: 'button:has-text("Start blank")',
+};
+
+/**
+ * Represents a Template Part block in the full site editor.
+ */
+export class TemplatePartBlock {
+	static blockName = 'Template Part';
+	static blockEditorSelector = '[aria-label="Block: Template Part"]';
+
+	private page: Page;
+	block: Locator;
+
```

Comment:
This is the new pattern -- we pass in a `Locator` to the top of the block, instead of an `ElementHandle`